packages/in_app_purchase/example/lib/main.dart

```diff
@@ -0,0 +1,372 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// ignore_for_file: avoid_print
+
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:in_app_purchase/in_app_purchase.dart';
+import 'package:in_app_purchase_tizen/billing_manager_wrappers.dart';
+import 'package:in_app_purchase_tizen/in_app_purchase_tizen.dart';
+
+void main() {
+  WidgetsFlutterBinding.ensureInitialized();
+
+  InAppPurchaseTizenPlatform.registerPlatform();
+  runApp(_MyApp());
+}
+
+// To try without auto-consume, change `true` to `false` here.
+const bool _kAutoConsume = true;
+
+/// The [_kConsumableType]'s value:
+/// "1": CONSUMABLE
+/// "2": NON-CONSUMABLE
+/// "3": LIMITED-PERIOD
+/// "4": SUBSCRIPTION
+const int _kConsumableType = 1;
+const String _kAppId = '3201504002021';
+const String _kCustumId = '810000047372';
+const String _kServerType = 'DEV';
+const String _kCountryCode = 'US';
+const String _kItemType = '2';
+const String _kPageSize = '20';
+const String _kPageNum = '1';
+// ignore: unused_element
+const String _kSecurityKey = '7db0YeFxHPtajwvVShBC5auI4iSJraKy8yhPZe13yvM=';
+
+/// The check value is used by the DPI service to verify Purchase List and Products List API requests.
+/// It is a Base64 hash generated by applying the HMAC SHA256 algorithm on a concatenated string of parameters using the DPI security key.
+///
+/// The [_kProductCheckValue] is generated with [_kAppId] and [_kCountryCode] and [_kSecurityKey];
+/// var hmacSha256 = Hmac(sha256, utf8.encode(_kSecurityKey));
+/// String _kProductCheckValue = base64Url.encode(hmacSha256.convert(utf8.encode(_kAppId + _kCountryCode)).bytes);
+String _kProductCheckValue = 'sbKzQJrFdeLkfXTQUdjjq4jt7ykx79QmJno9ZLam+SM=';
+
+/// The [_kPurchaseCheckValue] is generated with [_kAppId] and [_kCustumId] and [_kCountryCode] and [_kItemType] and [_kSecurityKey];
+/// var hmacSha256 = Hmac(sha256, utf8.encode(_kSecurityKey));
+/// String _kPurchaseCheckValue = base64Url.encode(hmacSha256.convert(utf8.encode(_kAppId + _kCustumId + _kCountryCode + _kItemType)).bytes);
+String _kPurchaseCheckValue = 'eAKHVR/FUfgEmDb3KmtgkozWAKyEb1OWXKdvkUV4km0=';
+
+// WARNING: Don't change order of `_kRequestParams`.
```

Comment:
> `queryProductDetails` expects an unordered set of identifiers

Yes, i konw, but in `billingManager.requestProducts(identifiers.toList())` , a `Set` of `identifiers` is converted to list, this must be Ordered.

Reply:
So I'm saying it's not the case for other platforms implementing the same API. If you really need to pass multiple parameters in a particular order to the API, it would be nice to minimize the number of the parameters to improve usability and avoid confusion as much as possible.

Reply:
I add a new api in addition file: setRequestParameters, this can pass all params that getproductlist, getpurchaselist and buyitem need. Just call it before call` queryProductDetails`, however, the set of `identifiers` will not be used in `billingManager.requestProducts`.
[Add tizen specific api:setRequestParameters](https://github.com/flutter-tizen/plugins/pull/562/commits/adf7c11f89b0f3705a7385e30b12cafd68fee506)

Reply:
`requestProducts`, `requestPurchases`, and `buyItem` are different APIs and they require different parameters. For example, `buyItem` only requires `appId` and `serverType` as parameters, but in your design the developer must pass parameters like `customId` and `itemType` even if they doesn't want to call `requestPurchases`. This doesn't make much sense and the developer will have a hard time guessing what `pageSize` means in that context since `pageSize` is not documented in the `BuyItem` documentation. Do you think this is still okay? Is the limitation intentional, because you couldn't find a better design?

Also you're passing `_kRequestParams` to `queryProductDetails` redundantly. You could pass an empty set if they are not used.

Reply:
Have modified the api.
In `setRequestParameters`, only `appId` is required, users can pass the parameters they want. If miss some param, an error will be throw by native side.